units/crons.go

```diff
@@ -718,8 +718,10 @@ func PopulateAgentMonitorDeployJobs(env evergreen.Environment) amboy.QueueOperat
 
 }
 
-// PopulateGenerateTasksJobs populates generate.tasks jobs for tasks that have started running their generate.tasks command.
-func PopulateGenerateTasksJobs(env evergreen.Environment) amboy.QueueOperation {
+// PopulateFallbackGenerateTasksJobs populates generate.tasks jobs for tasks that have started running their generate.tasks command.
+// Since the original generate.tasks request kicks off a job immediately, this function only serves as a fallback in case the
+// original job fails to run. If the original job succeeds, the jobs created here will no-op.
```

Comment:
This comment is mostly right, but the job shouldn't be created if the job already completed (with success or error) - the cron populator onlys scrapes up those tasks that still have to generate tasks, because `GenerateNotRun` evaluates to [false after a job tries to generate and save the generated tasks](https://github.com/evergreen-ci/evergreen/blob/638e37f61359d4a5357669a3ff99de6d170a9af7/units/generate_tasks.go#L299-L303).

Reply:
Yeah, I meant it would no-op if if gets kicked off before the first job finished. Updated the comment

Reply:
Yeah, I meant it would no-op if if gets kicked off before the first job finished. Updated the comment

Reply:
The job supposedly shouldn't be created if it's already in the queue though because there's a scope on the job (which prevents duplicate jobs for the same task from being in the queue simultaneously). Did you see a case in your tests that indicated that there were two jobs in the queue for the same task at the same time?